lib/mongoid/attributes.rb

```diff
@@ -362,5 +360,11 @@ def lookup_attribute_presence(name, value)
       end
       value.present?
     end
+
+    def __read_attribute__(name, field, raw)
+      value = field ? field.demongoize(raw) : raw
+      attribute_will_change!(name) if value.resizable?
+      value
+    end
```

Comment:
Now this method doesn't read any attributes, contrary to its  name.

I suggest leaving the existing code to be as it is now.

Reply:
Let me think of a better name for this method.